src/compute/src/compute_state.rs

```diff
@@ -156,10 +156,18 @@ impl<'a, A: Allocate> ActiveComputeState<'a, A> {
 
             // Initialize frontiers for each object, and optionally log their construction.
             for (object_id, collection_id) in exported_ids {
-                self.compute_state.reported_frontiers.insert(
+                if let Some(frontier) = self.compute_state.reported_frontiers.insert(
                     object_id,
                     Antichain::from_elem(timely::progress::Timestamp::minimum()),
-                );
+                ) {
+                    error!(
+                        "existing frontier {} for newly created dataflow id {}",
+                        frontier
+                            .get(0)
+                            .expect("empty frontier found for newly created dataflow"),
```

Comment:
Good point, I agree. I think that an error! here is enough. There would definitely be a bug if a frontier were in the metadata at this point, but I think it would be too drastic to force a `computed` crash and rehydration cycle because of it.